x/oracle/keeper/keeper.go

```diff
@@ -168,16 +168,20 @@ func (k Keeper) IterateExchangeRates(ctx sdk.Context, handler func(string, sdk.D
 
 // GetFeederDelegation gets the account address to which the validator operator
 // delegated oracle vote rights.
-func (k Keeper) GetFeederDelegation(ctx sdk.Context, operator sdk.ValAddress) sdk.AccAddress {
-	store := ctx.KVStore(k.storeKey)
+func (k Keeper) GetFeederDelegation(ctx sdk.Context, operator sdk.ValAddress) (sdk.AccAddress, error) {
+	// check that the given validator exists
+	if val := k.StakingKeeper.Validator(ctx, operator); val == nil || !val.IsBonded() {
```

Comment:
Do we need to check against the staking module here? Can we just check against our oracle store? I.e. get the feeder delegation if that is empty or errors, return an error.

Reply:
The reason we do it this way is so that validators are delegated to themselves by default : 
```go
	bz := store.Get(types.GetFeederDelegationKey(operator))
	if bz == nil {
		// by default the right is delegated to the validator itself
		return sdk.AccAddress(operator), nil
	}
```
If we errored instead here, before a validator could vote, they'd have to delegate to themselves first

Reply:
Right, but if `operator` doesn't exist in state, we can return an error. That's what I meant.